File: Practica5/Marco teorico/3. Destructor/C. Invocacion implicita de destructores/3.3.py
class MiClase:
    def __init__(self, n):
        self.nombre = n
        print("Constructor llamado para", self.nombre)

    def mostrarNombre(self):
        print("Nombre:", self.nombre)

    def __del__(self):
        print("Destructor llamado para", self.nombre)

objeto = MiClase("Ejemplo")
objeto.mostrarNombre()

File: Practica5/Marco teorico/3. Destructor/C. Invocacion implicita de destructores/3.3.cpp
#include <iostream>
#include <string>


using namespace std;


class MiClase {
private:
    string nombre;


public:
    // Constructor
    MiClase(string n) : nombre(n) {
        cout << "Constructor llamado para " << nombre << endl;
    }


    // Destructor
    ~MiClase() {
        cout << "Destructor llamado para " << nombre << endl;
    }


    // Método para mostrar el nombre
    void mostrarNombre() {
        cout << "Nombre: " << nombre << endl;
    }
};

int main() {
    {
        // Crear un objeto de la clase MiClase
        MiClase objeto("Ejemplo");


        // Llamar al método para mostrar el nombre
        objeto.mostrarNombre();


    } // Al llegar al final del ámbito, el destructor de 'objeto' se invoca automáticamente


    return 0;
}
